src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,157 @@
+import React, {
+  FC, useContext, useEffect, useMemo, useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { deleteTodos, patchTodos } from '../../api/todos';
+import { Loader } from '../Loader';
+import { TodoContext } from '../TodoContext';
+
+import './todo-item.css';
+
+type Props = {
+  todo: Todo
+};
+
+export const TodoItem: FC<Props> = React.memo(
+  ({
+    todo,
+  }) => {
+    const { completed, title, id } = todo;
+
+    const [isLoading, setIsLoading] = useState(false);
+    const [isEditing, setIsEdit] = useState(false);
+    const [inputValue, setIsInputValue] = useState(title);
+
+    const {
+      isClearAllCompleted,
+      isToggleAllCompleted,
+      isToggleAllActive,
+      todosGetter,
+      setIsDeleteError,
+      setIsPostError,
+    } = useContext(TodoContext);
+
+    const isClearAllActive = useMemo(
+      () => isClearAllCompleted && completed,
+      [completed, isClearAllCompleted],
+    );
+
+    const setFocusOnForm = () => {
+      const inputElement
+      = document
```

Comment:
Avoid using direct DOM queries in a React component. You should use Refs to interact with DOM elements.

Reply:
Fixed